### PR TITLE
Re-Implement subscription filters in Go

### PR DESCRIPTION
**Pull request by jirevwe**

### CODE REVIEW

File: pkg/compare/compare.go [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+)
+
+type CompareFunc func(x, y interface{}) bool
+
+func defaultCompareMap() map[string]CompareFunc {
```
Comment (Subomi) [inline pkg/compare/compare.go:11]: @jirevwe Can this be a const? Can't think of a scenario where this will need to change.

Comment (jirevwe) [reply pkg/compare/compare.go:11]: Oh yeah

Comment (jirevwe) [reply pkg/compare/compare.go:11]: Hmm, apparently maps can't be made constants, I tried it and it didn't work.

Comment (Subomi) [reply pkg/compare/compare.go:11]: Maybe variable, then?

**Review by Subomi — APPROVED** [review]

**Pull request merged**